projects/igniteui-angular/src/lib/accordion/accordion.component.ts

```diff
@@ -0,0 +1,324 @@
+            }
+        });
+    }
+
+    /**
+     * Collapses all expanded expansion panels.
+     *
+     * ```typescript
+     * accordion.collapseAll();
+     * ```
+     */
+    public collapseAll() {
+        this.panels.forEach(panel => {
+            if (!panel.collapsed && !panel.header.disabled) {
+                panel.collapse();
+            }
+        });
+    }
+
+    public handleKeydown(event: KeyboardEvent, panel: IgxExpansionPanelComponent) {
+        const key = event.key.toLowerCase();
+        if (!(ACCORDION_NAVIGATION_KEYS.has(key))) {
+            return;
+        }
+        // TO DO: if we ever want to improve the performance of the accordion,
+        // enabledPanels could be cached (by making a disabledChange emitter on the panel header)
+        this.enabledPanels = this._panels.filter(p => !p.header.disabled);
+        event.preventDefault();
+        if (event.repeat) {
+            setTimeout(() => this.handleNavigation(event, panel), 1);
+        } else {
+            this.handleNavigation(event, panel);
+        }
+    }
+
+    private handleNavigation(event: KeyboardEvent, panel: IgxExpansionPanelComponent) {
+        switch (event.key.toLowerCase()) {
+            case 'home':
+                this.enabledPanels[0].header.innerElement.focus();
+                break;
+            case 'end':
+                this.enabledPanels[this.enabledPanels.length - 1].header.innerElement.focus();
+                break;
+            case 'arrowup':
+            case 'up':
+                this.handleUpDownArrow(true, event, panel);
+                break;
+            case 'arrowdown':
+            case 'down':
+                this.handleUpDownArrow(false, event, panel);
+                break;
+            default:
+                return;
+        }
+    }
+
+    private handleUpDownArrow(isUp: boolean, event: KeyboardEvent, panel: IgxExpansionPanelComponent): void {
+        if (!event.altKey) {
+            const focusedPanel = panel;
+            const next = this.getNextPanel(focusedPanel, isUp ? -1 : 1);
+            if (next === focusedPanel) {
+                return;
+            }
+            next.header.innerElement.focus();
+        }
+        if (event.altKey && event.shiftKey) {
+            if (isUp) {
+                this.collapseAll();
+            } else {
+                this.expandAll();
+            }
+        }
+    }
+
+    private getNextPanel(panel: IgxExpansionPanelComponent, dir: 1 | -1 = 1): IgxExpansionPanelComponent {
+        const panelIndex = this.enabledPanels.indexOf(panel);
+        return this.enabledPanels[panelIndex + dir] || panel;
+    }
+
+    private subToChanges() {
+        this.unsubChildren$.next();
+        this._panels.forEach(panel => {
+            panel.contentExpanded.pipe(takeUntil(this.unsubChildren$)).subscribe((args: IExpansionPanelEventArgs) => {
+                this.expandedPanels.add(args.owner);
+                this.panelExpanded.emit(args);
```

Comment:
![image](https://user-images.githubusercontent.com/47036441/123107663-6a394400-d442-11eb-82f7-500537f41966.png)
We have implemented the event this way because of the specification
https://github.com/IgniteUI/igniteui-angular/wiki/Accordion-Specification#events

Reply:
Yes, I've seen the spec but I still think this should be changed. Just look at any of our controls which uses the overlay service and you'll see what I mean. Take the `igxToggle` for example - it essentially reemits the same events from the overlay service but it changes the `owner`. That is because the developer who will handle the event, will attach their handler to the toggle directive and from their point of view the toggle directive was the one who emitted the event, not the overlay service. 

And it makes sense to expect that the `owner` of the event is the control that you've attached your handler to since it was the one who actually emitted it. So in this case I think that we may want to revise this section of the specification, considering it's been signed off. 

@kdinev, can you share your thoughts on this?

Reply:
I'll let @damyanpetev share his thoughts and if you both agree the spec needs to change, then go ahead!

Reply:
True the toggle/drop down re-emit, however they do that _internally_ and and out users are not even necessarily aware of that, thus the owner swap :) There's no precedence I can think of for such re-emit.

I too assume `owner` is the reference to the component that emits the event and that's true for other components in the product. I understand there's very little API to interact with accordion itself and it's easier to re-emit as is, but that `owner` may be confusing for those already used to it being one way.

The use of those events is kind of limited, considering they'll barely (if any) benefit if the panels are repeated; as in `<igx-accordion (panelExpanding)="handler()>` is identical to `<igx-expansion-panel *ngFor=".." (expanding)="handler()">`, right? So this is only a convenience for the cases where each panel is manually declared, yet some common handling is needed, which should likely be reflected in the samples. Just my 2c

PS: I was almost right about the no precedence... however I went looking and figured the Hierarchical Grid re-emits a lot of events, and sure enough - inherited grid events do emit with the grid as owner intentionally, so we've already strayed from the rule. I'm still going to count that as an exception for the HGrid, and still lean towards being consistent elsewhere :)